extensions/picocli/deployment/src/main/java/io/quarkus/picocli/deployment/PicocliProcessor.java

```diff
@@ -99,6 +104,19 @@ void picocliRunner(ApplicationIndexBuildItem applicationIndex,
             additionalBean.produce(AdditionalBeanBuildItem.unremovableOf(DefaultPicocliCommandLineFactory.class));
             quarkusApplicationClass.produce(new QuarkusApplicationClassBuildItem(PicocliRunner.class));
         }
+
+        // make the arguments of @Command unremovable beans if necessary
+        Collection<AnnotationInstance> commandInstances = index.getAnnotations(CommandLine.Command.class);
+        for (AnnotationInstance commandInstance : commandInstances) {
+            AnnotationValue versionProvider = commandInstance.value("versionProvider");
```

Comment:
so should we lookup all of the names referenced in #38077 and fix it for all of them?

```
converter
completionCandidates parameterConsumer
preprocessor
converter
completionCandidates
parameterConsumer
preprocessor
versionProvider
defaultValueProvider
modelTransformer
preprocessor
```

?

Reply:
I would prefer we do it on a case by case basis as doing it too broadly could cause problems (mostly because we none of us knows what those are used for)

Reply:
well, I do :) 

These are all config options on picocli commands where picocli uses IFactory to instantiate them.

We provide IFactory to have these created - and just like VersionProvider here if user don't put @Unremovable on them cannot configure them via CDI beans and MUST add @Unremovable to have it to work.

afaics since we wont register them as unremovable *unless* they are referenced by users picocli and if they don't make that class a @Singleton or some scope then it will just instantiate as a regular instance - so no difference in performance/impact unless user actually does make a bean ?

Reply:
I know we do provide a factory :). I guess reading it's docs it's safe to say that it is always used for the classes you mention.

Reply:
PR updated